elifetools/xmlio.py

```diff
@@ -7,21 +7,36 @@
 xmlio can do input and output of XML, allowing it to be edited using ElementTree library
 """
 
+class CustomXMLParser(ElementTree.XMLParser):
+    doctype_dict = {}
+    def doctype(self, name, pubid, system):
+        self.doctype_dict["name"] = name
+        self.doctype_dict["pubid"] = pubid
+        self.doctype_dict["system"] = system
+
 def register_xmlns():
     """
     Register namespaces globally
     """
     ElementTree.register_namespace("mml","http://www.w3.org/1998/Math/MathML")
     ElementTree.register_namespace("xlink","http://www.w3.org/1999/xlink")
+    ElementTree.register_namespace("ali","http://www.niso.org/schemas/ali/1.0/")
 
-def parse(filename):
-    parser = ElementTree.XMLParser(html=0, target=None, encoding='utf-8')
+def parse(filename, return_doctype_dict=False):
```

Comment:
from the outside world this looks backwards compatible, it is always using `CustomXMLParser` (also in the backward compatible case) but it seems it doesn't change the behavior